src/Native/Runtime/arm/AllocFast.S

```diff
@@ -2,115 +2,464 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
-#include <unixasmmacros.inc>
-
 .syntax unified
 .thumb
 
-#ifdef _DEBUG
-.rodata
-AssertMsg:                    .asciz "__FILE__:%s: %s is not implemented\n"
-FileName:                     .asciz "AllocFast.S"
-RhpNewFastName:               .asciz "RhpNewFast"
-RhpNewFinalizableName:        .asciz "RhpNewFinalizable"
-RhpNewObjectName:             .asciz "RhpNewObject"
-RhpNewArrayName:              .asciz "RhpNewArray"
-RhpNewArrayRareName:          .asciz "RhpNewArrayRare"
-RhpNewFastAlign8Name:         .asciz "RhpNewFastAlign8"
-RhpNewFinalizableAlign8Name:  .asciz "RhpNewFinalizableAlign8"
-RhpNewFastMisalignName:       .asciz "RhpNewFastMisalign"
-RhpNewArrayAlign8Name:        .asciz "RhpNewArrayAlign8"
-
-.text
-.macro GEN_ASSERT_FUNC func
-        GEN_ASSERT AssertMsg, FileName, \func
-.endm
-#endif
-
-// TODO: Implement Arm support
+#include <unixasmmacros.inc>
+#include <AsmOffsets.inc>         // generated by the build from AsmOffsets.cpp
 
 // Allocate non-array, non-finalizable object. If the allocation doesn't fit into the current thread's
 // allocation context then automatically fallback to the slow allocation path.
 //  r0 == EEType
-//
-//    LEAF_ENTRY RhpNewFast, _TEXT
-//#ifdef _DEBUG
-//        GEN_ASSERT_FUNC RhpNewFastName
-//#endif
-//    LEAF_END RhpNewFast, _TEXT
-//
+LEAF_ENTRY RhpNewFast, _TEXT
+        PROLOG_PUSH "{r4,lr}"
+        mov	        r1, r0
```

Comment:
`r1` is scratch register. Is it written somewhere that `INLINE_GETHREAD` will preserve it?